src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -685,7 +685,7 @@ private ImmutableArray<TypeParameterSymbol> LoadTypeParameters(ref DiagnosticInf
             }
         }
 
-        public override ImmutableArray<TypeSymbolWithAnnotations> TypeArguments => IsGenericMethod ? TypeParameters.SelectAsArray(TypeMap.AsTypeSymbolWithAnnotations) : ImmutableArray<TypeSymbolWithAnnotations>.Empty;
+        public override ImmutableArray<TypeSymbolWithAnnotations> TypeArguments => IsGenericMethod ? GetTypeParametersAsTypeArguments() : ImmutableArray<TypeSymbolWithAnnotations>.Empty;
```

Comment:
Will we need to apply NonNullTypes context here at some point? If so, maybe add a PROTOTYPE

Reply:
`GetTypeParametersAsTypeArguments()` passes a NonNullTypes context.

---
In reply to: [203874496](https://github.com/dotnet/roslyn/pull/28687#discussion_r203874496) [](ancestors = 203874496)